### ISSUE TITLE
use-destructured-vars

### ISSUE THREAD

**Issue by hornta** [eslint/eslint#11215]
**Please describe what the rule should do:**
It should report on when using a variable in dot notation from a previously destructured statement.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// 👎
const { foo } = bar;

bar.foo(); // `foo` has already been destructured from bar. 
```

```js
// 👍
const { foo } = bar;

foo();
```

### Options:
- consistent true/false

#### "use-destructured-vars": ["error", { consistent: true }}
```js
// 👎
const { foo } = bar;

bar.xyz(); // `xyz` can be destrucutured from `bar` because `bar` has previously been destructured from
```

```js
// 👍
const foo = bar.foo;

bar.xyz();
```

#### "use-destructured-vars": ["error", { consistent: false }}
```js
// 👍
const { foo } = bar;

bar.xyz(); // allowed to use xyz in this form even if bar has previously been destructured from.
```

**Why should this rule be included in ESLint (instead of a plugin)?**
It seems reasonable.

**Are you willing to submit a pull request to implement this rule?**
Not right now because I do not have the time, but later for sure!

**Comment by hornta** [comment]
Bump
